tools/slang-cpp-extractor/parser.cpp

```diff
@@ -1833,7 +1838,36 @@ SlangResult Parser::_maybeParseContained(Node** outNode)
             fieldNode->m_name = nameToken;
             fieldNode->m_reflectionType = m_currentScope->getContainedReflectionType();
             fieldNode->m_isStatic = isStatic;
-
+            if (fieldNode->m_reflectionType == ReflectionType::Reflected)
+            {
+                static const char* illegalTypes[] = {
+                    "size_t",
+                    "Int",
+                    "UInt",
+                    "Index",
+                    "Count",
+                    "UIndex",
```

Comment:
How will this code get updated if the names these types change?

Reply:
There is not a good way to guard against that unfortunately. But we shouldn't be changing the names of these types.